Added date_consider_month/year and CI tests

Rethous opened the pull request:
I added the possibility to create all **date_consider/1** for all dates in a given month or a whole year. Also added CI tests for those, to test if _'illegal'_ dates get created.

Commit:
added date_consider_month/year, last_date_month/year, leap year calculation

Commit:
added date_consider_month/year, last_date_month/year, leap year calculation

Commit:
Added date_consider for whole month and years.

Commit:
Clean up and improvement of calc_leap_year, date_consider_month/year and last_date_month/year
Committer: moritz_igel <igel1@uni-potsdam.de>

Commit:
Rework of last_date_month and some more cleanup/optimisation

Commit:
removed unnessecery line in date-lib and added tests

Commit:
moved test files in own folder

Commit:
Fixed Assertionerror for 30.4.2021

Commit:
Renamed test files

Rethous committed:
Create .github

Rethous committed:
Delete .github

Rethous committed:
Create .github

Commit:
Added workflow for CI

Rethous committed:
Delete .github

Commit:
Merge branch 'test' of https://github.com/Rethous/asp-date-lib
changes through github

Commit:
Renamed test to tests

Commit:
renamed CI job  name

Commit:
changed run command of CI

Commit:
added clingo check for CI

Commit:
run command change

Commit:
added pythonpath job

Commit:
Chnged set comparison method for 'a not in answer'

Commit:
Changed test methode when 'not in answer' is expected

Commit:
Added date_consider_month/2 and date_consider_year/1 description

Commit:
Removed test jobs that were used to try to fix module import bug

sthiele commented on README.md:
```diff
@@ -4,15 +4,35 @@
+
 ## date_consider/1( (D,M,Y) )
 
+It is recommended to use date_consider_month/2 or date_consider_year/1, because it prevents the accidental creation of 'illegal' dates i.e. 29th of February in a non leap year.
```
Maybe better.
:warning: Do not use!  It allows the creation of  *illegal* dates i.e. 29th of February in a non leap year.
It is recommended to use date_consider_month/2 or date_consider_year/1

sthiele replied:
@MaxOstrowski  Maybe we should remove date_consider/1 from the README at all. If it should not be used as Input predicate. I wish we could make predicates private.

sthiele replied:
Maybe add integrity constraints for *illegal* dates? or a save alternative like `checked_date_consider`?

sthiele commented on date-lib.lp:
```diff
@@ -134,6 +134,52 @@ date_relative((D,M,Y),Julian-OJulian) :-
+%
 %
 % Testcases
 %
```
I think the test cases can now be removed from the encoding.

sthiele commented on tests/test_date.py:
```diff
@@ -0,0 +1,55 @@
+import clingo
+import sys
+
+class MyModel:
```
Is this class needed?
- cost is never used

sthiele commented on tests/test_date.py:
```diff
@@ -0,0 +1,55 @@
+
+def run(instance):
+    ctl = clingo.Control()
+    m = MyModel()
```
As `answer` is always turned into a set just use a set here.

sthiele replied:
```
m = {}
def on_model(model):
       m = set(model.symbols(atoms=True))
```

sthiele reviewed (commented):
Thanks for the PR :)

sthiele commented:
The encoding contains an old duplicate of the documenation. These will be hard to keep in synce. I recommend to remove it.
https://github.com/potassco/asp-date-lib/blob/85f54561d14845763d4f5fc1433c5a75d2553f8a/date-lib.lp#L1-L70

sthiele commented on date-lib.lp:
```diff
@@ -134,6 +134,52 @@ date_relative((D,M,Y),Julian-OJulian) :-
+	is_leap_year(YYYY).
+
+%
+% date_consider given month
```
Improve comment.

sthiele commented on date-lib.lp:
```diff
@@ -134,6 +134,52 @@ date_relative((D,M,Y),Julian-OJulian) :-
+	last_date_month(DD,MM,YYYY).
+
+%
+% date_consider given year
```
Improve comment.

MaxOstrowski commented:
Thanks for your comments @sthiele.
The plan is to mark `date_consider` as deprecated and to remove it from the readme otherwise.

Commit:
Cleanup of comments

Commit:
Fixed display error for date_consider/1

MaxOstrowski commented on README.md:
```diff
@@ -4,14 +4,35 @@
+## (deprecated) date_consider/1( (D,M,Y) )
+
+Do not use! It allows the creation of illegal dates i.e. 29th of February in a non leap year.
+It is recommended to use date_consider_month/2 or date_consider_year/1
```
missing dot

MaxOstrowski commented:
last_date_month is not documented as an output predicate ?

Commit:
Spelling fix and added last_date_month/3

Commit:
Removed class 'MyModel'

Rethous commented:
I believe I have addressed all comments and made the proper changes.
About marking date_consider/1 as deprecated. I was think about moving it to the output predicates, since that would describe its current role more accurately. 
What do you think? @sthiele @MaxOstrowski

MaxOstrowski commented:
Actually sounds like a very good idea. I actually needed that already.
Do add the description as output predicate and I will merge.
Thanks a lot!

Commit:
Moved date_consider/1 to output predicate
